src/main/java/teammates/ui/newcontroller/PostCourseEnrollSaveAction.java

```diff
@@ -0,0 +1,200 @@
+
+            return new JsonResult("The enrollment failed, possibly because some students were re-enrolled before "
+                            + "the previous enrollment action was still being processed by TEAMMATES database "
+                            + "servers. Please try again after about 10 minutes. If the problem persists, "
+                            + "please contact TEAMMATES support", HttpStatus.SC_NOT_FOUND);
+        }
+    }
+
+    private List<StudentAttributes>[] enrollAndProcessResultForDisplay(String studentsInfo, String courseId)
+            throws EnrollException, EntityDoesNotExistException, InvalidParametersException, EntityAlreadyExistsException {
+        CourseEnrollmentResult enrollResult = logic.enrollStudents(studentsInfo, courseId);
+        List<StudentAttributes> students = enrollResult.studentList;
+
+        // Adjust submissions for all feedback responses within the course
+        List<FeedbackSessionAttributes> feedbackSessions = logic.getFeedbackSessionsForCourse(courseId);
+        for (FeedbackSessionAttributes session : feedbackSessions) {
+            // Schedule adjustment of submissions for feedback session in course
+            taskQueuer.scheduleFeedbackResponseAdjustmentForCourse(
+                    courseId, session.getFeedbackSessionName(), enrollResult.enrollmentList);
+        }
+
+        students.sort(Comparator.comparing(obj -> obj.updateStatus.numericRepresentation));
+
+        return separateStudents(students);
+    }
+
+    /**
+     * Separate the StudentData objects in the list into different categories based
+     * on their updateStatus. Each category is put into a separate list.
+     *
+     * @return An array of lists of StudentAttributes objects in which each list contains
+     *         student with the same updateStatus
+     */
+    @SuppressWarnings("unchecked")
+    private List<StudentAttributes>[] separateStudents(List<StudentAttributes> students) {
+
+        ArrayList<StudentAttributes>[] lists = new ArrayList[StudentUpdateStatus.STATUS_COUNT];
+        for (int i = 0; i < StudentUpdateStatus.STATUS_COUNT; i++) {
+            lists[i] = new ArrayList<>();
+        }
+
+        for (StudentAttributes student : students) {
+            lists[student.updateStatus.numericRepresentation].add(student);
+        }
+
+        for (int i = 0; i < StudentUpdateStatus.STATUS_COUNT; i++) {
+            StudentAttributes.sortByNameAndThenByEmail(lists[i]);
+        }
+
+        return lists;
+    }
+
+    private List<EnrollResultPanel> getInstructorCourseEnrollResult(List<StudentAttributes>[] students) {
+        List<EnrollResultPanel> enrollResultPanelList = new ArrayList<>();
+
+        for (int i = 0; i < StudentUpdateStatus.STATUS_COUNT; i++) {
+            String panelClass = "";
+
+            switch (StudentUpdateStatus.enumRepresentation(i)) {
+            case ERROR :
+                panelClass = "bg-danger";
```

Comment:
For better separation of concern, let's not have HTML classes in Java code anymore.

Reply:
@wkurniawan07 Let's change them to strings e.g. `danger`, `warning` and then apply a pipe to transform it to a css class in [here](https://github.com/TEAMMATES/teammates/pull/9329/files/291cc70d201d0c5ee3d115b0acfb6fdde48310b7#diff-35b02ff0aa8eefda8e49074c5ed8fc94R237)?